src/lib.rs

```diff
@@ -141,42 +133,16 @@ pub fn run() -> Result<Vec<Warning>, Box<dyn Error>> {
     Ok(warnings)
 }
 
-fn get_lines(fe: FileEntry) -> io::Result<Vec<LineEntry>> {
-    let mut f = File::open(&fe.path)?;
+fn get_line_entries(fe: FileEntry, lines: Vec<String>) -> io::Result<Vec<LineEntry>> {
```

Comment:
```suggestion
fn get_line_entries(fe: FileEntry, lines: Vec<String>) -> Vec<LineEntry> {
```
I think we can omit `io::Result`.